src/main/java/com/ge/predix/uaa/token/lib/FastTokenServices.java

```diff
@@ -85,29 +87,43 @@
 
     private List<String> trustedIssuers;
 
-    private Map<String, SignatureVerifier> tokenKeys;
+    private Cache<String, SignatureVerifier> tokenKeys;
 
     /**
      * Creates the FastTokenServices with {@link FastTokenServices#DEFAULT_TTL_24HR_MILLIS}.
      */
-     public FastTokenServices() {
-         initTokenKeysMap(DEFAULT_TTL_24HR_MILLIS);
-     }
+    public FastTokenServices() {
+        initTokenKeysMap(DEFAULT_TTL_24HR_MILLIS);
+    }
 
-     /**
-      * @param issuerPublicKeyTTLMillis A value of -1 implies cache never expires.
-      */
-     public FastTokenServices(final long issuerPublicKeyTTLMillis) {
-         initTokenKeysMap(issuerPublicKeyTTLMillis);
-     }
+    /**
+     * @param issuerPublicKeyTTLMillis A value of -1 implies cache never expires.
+     */
+    public FastTokenServices(final long issuerPublicKeyTTLMillis) {
+        initTokenKeysMap(issuerPublicKeyTTLMillis);
+    }
 
     @Override
     public void afterPropertiesSet() throws Exception {
         initTokenKeysMap(issuerPublicKeyTTLMillis);
     }
 
     private void initTokenKeysMap(final long ttlMillis) {
-        this.tokenKeys = new PassiveExpiringMap<>(ttlMillis);
+         /* Memory leak was identified in performance tests and defect is added with id DE68869. This check is added to
+        address a memory leak possible through lazy expiration mechanism in passive expiring map.
+        Though we are initializing passive expiring map with time to expire the entries. The entries are not flushed
+        unless there is a get call made on the expired entries.
```

Comment:
Minor.  Does it reads better?

"Memory leak was identified in performance tests due to a lazy expiration mechanism in passive expiring map. 
 Even though we were initializing passive expiring map with TTL, the entries were not flushed unless a get() call is made on the expired entries or a call that prompts entire map scan, for example size().  This could lead to a memory leak as expired entry might never be retrieved.

For instance, when loadAuthentication call is made it adds the signature verifier for the token issuer in tokenKeys
map.  This entry is not removed upon expiration unless another loadAuthentication call made after the expired time with the access token from the same issuer.  Abandoned issuers will be stuck in the cache.

We have decide to use caffeine as the caching solution to address the memory leak."

Reply:
+1